flow360/component/simulation/meshing_param/params.py

```diff
@@ -0,0 +1,57 @@
+from typing import List, Literal, Optional, Union
+
+import pydantic as pd
+from edge_params import SurfaceEdgeRefinement
+from face_params import FaceRefinement
+from volume_params import CylindricalRefinement, ZoneRefinementTypes
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.framework.unique_list import UniqueItemList
+
+
+class MeshingParameters(Flow360BaseModel):
+    """
+    Meshing parameters for volume and/or surface mesher.
+
+    In `Simulation` this only contains what the user specifies. `Simulation` can derive and add more items according to other aspects of simulation. (E.g. BETDisk volume -> ZoneRefinement)
+
+    Meshing related but may and maynot (user specified) need info from `Simulation`:
+    1. Add rotational zones.
+    2. Add default BETDisk refinement.
+
+    Affects volume meshing:
+    - farfield
+    - refinement_factor
+    - gap_treatment_strength
+    - `class` BoundaryLayerRefinement
+    - `class` UniformRefinement
+    - `class` CylindricalRefinement
+
+    Affects surface meshing:
+    - surface_layer_growth_rate
+    - `class` FaceRefinement
+    - `class` SurfaceEdgeRefinement
+    """
+
+    # Volume **defaults**:
+    farfield: Literal["auto", "quasi-3d", "user-defined"] = pd.Field(
+        description="Type of farfield generation."
+    )
+    refinement_factor: pd.PositiveFloat = pd.Field(
```

Comment:
rename to global_refinement_factor? user might get confused with another keyword called refinements. also why do we need this to be seprate? can it be unified with refinments